File: android-exercises/5 - bottomBar/BottomBar_java/app/src/main/java/com/bugrahankaramollaoglu/bottombar_java/MainActivity.java
package com.bugrahankaramollaoglu.bottombar_java;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNav);

        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                if (item.getItemId() == R.id.homeItem) {
                    launchFragment(new HomeFragment());
                } else if (item.getItemId() == R.id.settingsItem) {
                    launchFragment(new SettingsFragment());
                } else if (item.getItemId() == R.id.bookmarksItem) {
                    launchFragment(new BookmarkFragment());
                }
                return true;
            }
        });


    }

    public void launchFragment(Fragment fragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        fragmentTransaction.replace(R.id.container, fragment);
        fragmentTransaction.commit();

        // you could also write it one-liner like
        // getSupportFragmentManager().beginTransaction().replace(R.id.container, fragment).commit();

    }
}

File: android-exercises/5 - bottomBar/BottomBar_kotlin/app/src/main/java/com/bugrahankaramollaoglu/bottombar_kotlin/MainActivity.kt
package com.bugrahankaramollaoglu.bottombar_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val homeFrag = HomeFragment()
        val bookmarkFrag = BookmarksFragment()
        val setFrag = SettingsFragment()

        val bottomNavigation = findViewById<BottomNavigationView>(R.id.bottomNavigation)

        bottomNavigation.setOnItemSelectedListener {

            when (it.itemId) {
                R.id.homeItem -> {
                    launcherFragment(HomeFragment())
                }

                R.id.bookmarksItem -> {
                    launcherFragment(BookmarksFragment())
                }

                R.id.settingsItem -> {
                    launcherFragment(SettingsFragment())
                }

            }

            true
        }

    }

    private fun launcherFragment(fragmentSwitch: Fragment) {
        supportFragmentManager.beginTransaction().apply {
            replace(R.id.fragmentContainer, fragmentSwitch).commit()
        }
    }
}